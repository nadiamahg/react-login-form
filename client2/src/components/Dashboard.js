import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { logoutUser } from "../actions/authActions";

export class Dashboard extends Component {

    onLogoutClick = e => {
        e.preventDefault();
        this.props.logoutUser();
        this.props.history.push("/"); // Redirect to Login page after logging out
    };


    render() {
        const { user } = this.props.auth;
        return (
            <div className="limiter">
                <div className="page-container">
                    <div className="wrap">
                        <span className="page-title"><b>You are logged in!</b></span>
                        <div className="user-info">
                            <div id="userEmail"><i className="material-icons user-icon">email</i><b>Email:</b> {user.email}</div>
                            <div id="userAge"><i className="material-icons user-icon">face</i><b>Age:</b> {user.age}</div>
                            <div id="userIsAdmin"><i className="material-icons user-icon">admin_panel_settings</i><b>isAdmin:</b> {user.isAdmin + ""}</div>
                        </div>
                        <div className="container-btn">
                            <button onClick={this.onLogoutClick} className="page-btn dashboard-container">
                                Logout
                             </button>
                         </div>
                
                    </div>
                </div>
          </div> 
        );
    }
}
Dashboard.propTypes = {
    logoutUser: PropTypes.func.isRequired,
    auth: PropTypes.object.isRequired
};
const mapStateToProps = state => ({
    auth: state.auth
});
export default connect(
    mapStateToProps, { logoutUser }
)(Dashboard);