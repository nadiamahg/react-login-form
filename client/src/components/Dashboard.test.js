import React from "react";
import { mount, configure, shallow } from "enzyme";
import {Dashboard} from "./Dashboard";
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

const mockLogoutfn = jest.fn();
const props = {
    auth : {
        user: {
            email: "test@gmail.com",
            age: "32",
            isAdmin: "false"
        }
    },
    logoutUser: mockLogoutfn
}

const mockHistory = { push: jest.fn() };

configure({ adapter: new Adapter() })

describe("Dashboard Component", () => {
    const wrapper = mount(<Dashboard {...props} history={mockHistory}/>)

    it('should render Dashboard correctly', () => {
        const component = renderer.create(<Dashboard {...props}/>,);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
    
    it('should have a logout button', () => {
        expect(wrapper.find('button').text()).toEqual('Logout');
    });

    it('should display user information', () => {
        expect(wrapper.find('div#userEmail').text()).toEqual('Email: test@gmail.com');
        expect(wrapper.find('div#userAge').text()).toEqual('Age: 32');
        expect(wrapper.find('div#userIsAdmin').text()).toEqual('isAdmin: false');
    });

    it('should call logout function when logout button clicked', () => {
        wrapper.find('button').simulate('click');
        expect(mockLogoutfn.mock.calls.length).toEqual(1);
    });

    it('should navigate to login form page when logout button clicked', () => {
        wrapper.find('button').simulate('click');
        expect(mockHistory.push.mock.calls[0]).toEqual(['/']);
    });
});