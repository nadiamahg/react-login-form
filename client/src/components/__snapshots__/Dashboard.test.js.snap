// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component should render Dashboard correctly 1`] = `
<div
  className="limiter"
>
  <div
    className="page-container"
  >
    <div
      className="wrap"
    >
      <span
        className="page-title"
      >
        <b>
          You are logged in!
        </b>
      </span>
      <div
        className="limiter"
      >
        <div
          id="userEmail"
        >
          <b>
            Email:
          </b>
           
          test@gmail.com
        </div>
        <div
          id="userAge"
        >
          <b>
            Age:
          </b>
           
          32
        </div>
        <div
          id="userIsAdmin"
        >
          <b>
            isAdmin:
          </b>
           
          false
        </div>
      </div>
      <div
        className="container-btn"
      >
        <button
          className="page-btn dashboard-container"
          onClick={[Function]}
        >
          Logout
        </button>
      </div>
    </div>
  </div>
</div>
`;
